### PR TITLE
OCC-245: Add the option to publish without Microsoft.SourceLink.GitHub

### PR DESCRIPTION
**Pull request by sarahelsaig**
[OCC-245](https://lombiq.atlassian.net/browse/OCC-245)
This intentionally targets `dev`, as it's not specific to OC 2.0.

Package publishing to Cloudsmith fails if Microsoft.SourceLink.GitHub was added to the project.

[OCC-245]: https://lombiq.atlassian.net/browse/OCC-245?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/actions/publish-nuget/action.yml [hunk 1]
```diff
@@ -114,7 +118,15 @@ runs:
...
-      run: Add-SourceLinkPackage
+      run: |
+        if ('${{ inputs.add-source-link-package }}' -eq 'true')
+        {
+          Add-SourceLinkPackage
+        }
+        else
+        {
+          dotnet restore -p:NuGetBuild=true
+        }
```
Comment (Piedone) [inline .github/actions/publish-nuget/action.yml]:
```suggestion
        if ('${{ inputs.add-source-link-package }}' -eq 'true')
        {
            Add-SourceLinkPackage
        }
        else
        {
            dotnet restore -p:NuGetBuild=true
        }
```

File: .github/actions/publish-nuget/action.yml [hunk 2]
```diff
@@ -114,7 +118,15 @@ runs:
     # This also implicitly restores NuGet dependencies.
     - name: Add Source Link package
       shell: pwsh
-      run: Add-SourceLinkPackage
+      run: |
+        if ('${{ inputs.add-source-link-package }}' -eq 'true')
+        {
+          Add-SourceLinkPackage
```
Comment (Piedone) [inline .github/actions/publish-nuget/action.yml]: How does this fail with Cloudsmith?

**Review by Piedone — CHANGES_REQUESTED** [review]

File: .github/actions/publish-nuget/action.yml (hunk 2 above)
Comment (sarahelsaig) [reply .github/actions/publish-nuget/action.yml]: Like this: https://github.com/Lombiq/Helpful-Libraries/actions/runs/9643961750/job/26595027509#step:4:288

> error: Package 'Microsoft.SourceLink.GitHub' is incompatible with 'all' frameworks in project '/home/runner/work/Helpful-Libraries/Helpful-Libraries/Lombiq.HelpfulLibraries.LinqToDb/Lombiq.HelpfulLibraries.LinqToDb.csproj'.

(All the other "Unable to resolve" errors in the log are irrelevant and they also went away after I turned off `Microsoft.SourceLink.GitHub`.)

Comment (Piedone) [reply .github/actions/publish-nuget/action.yml]: So it tries to restore all packages, as we observed earlier (hence the code in the script) but it only uses nuget.org and thus fails. Isn't this fixed by removing `--source` [here](https://github.com/Lombiq/GitHub-Actions/blob/dev/.github/actions/publish-nuget/Add-SourceLinkPackage.ps1#L48)? Only if this still works if there's no NuGet.config.

Comment (sarahelsaig) [reply .github/actions/publish-nuget/action.yml]: Perhaps. Do we know why was `--source 'https://api.nuget.org/v3/index.json'` necessary in the first place?

Comment (sarahelsaig) [reply .github/actions/publish-nuget/action.yml]: > Only if this still works if there's no NuGet.config.

Yes, that works without `--source`:
image

Comment (Piedone) [reply .github/actions/publish-nuget/action.yml]: Great, then we can just use that.

I think when writing that my understanding was that `--source` will use that source to restore SourceLink itself. If it works without it, then nothing else to do.

**Review by Piedone — APPROVED** [review]

**Pull request merged**